src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Components/ComponentMetadata.cs

```diff
@@ -126,6 +130,8 @@ public static class Component
         public const string FullyQualifiedNameMatch = "Components.FullyQualifiedNameMatch";
 
         public const string InitOnlyProperty = "Components.InitOnlyProperty";
+
+        public const string ConcreteContainingType = "Components.ConcreteContainingType";
```

Comment:
Consider not following the pattern that nothing is documented in Razor, and add a doc comment what these new metadata are used for.